src/v/v8_engine/tests/executor_test.cc

```diff
@@ -0,0 +1,178 @@
+
+    struct task_for_test_with_process_exception {
+        void operator()() { throw test_exception("Process exception"); }
+
+        void cancel() {}
+
+        void on_timeout() {}
+    };
+
+    struct task_for_test_with_process_set_cancel_exception {
+        void operator()() { sleep(1); }
+
+        void cancel() { throw test_exception("Set cancel() exception"); }
+
+        void on_timeout() {}
+    };
+
+    struct task_for_test_with_process_set_on_timeout_exception {
+        void operator()() { sleep(1); }
+
+        void cancel() {}
+
+        void on_timeout() {
+            throw test_exception("Set on_timeout() exception");
+        }
+    };
+
+    v8_engine::executor test_executor(1, ss::smp::count);
+
+    auto fut_process = test_executor.submit(
+      task_for_test_with_process_exception(), std::chrono::milliseconds(20));
+
+    BOOST_REQUIRE_EXCEPTION(
+      fut_process.get(), test_exception, [](const test_exception& e) {
+          return "Process exception" == std::string(e.what());
+      });
+
+    auto fut_cancel = test_executor.submit(
+      task_for_test_with_process_set_cancel_exception(),
+      std::chrono::milliseconds(20));
+
+    BOOST_REQUIRE_EXCEPTION(
+      fut_cancel.get(), test_exception, [](const test_exception& e) {
+          return "Set cancel() exception" == std::string(e.what());
+      });
+
+    auto fut_on_timeout = test_executor.submit(
+      task_for_test_with_process_set_on_timeout_exception(),
+      std::chrono::milliseconds(20));
+
+    BOOST_REQUIRE_EXCEPTION(
+      fut_on_timeout.get(), test_exception, [](const test_exception& e) {
+          return "Set on_timeout() exception" == std::string(e.what());
+      });
+
+    test_executor.stop().get();
+}
+
+SEASTAR_THREAD_TEST_CASE(stop_executor_with_finish_item_test) {
+    struct task_for_test {
+        explicit task_for_test(char& is_finish)
+          : _is_finish(is_finish) {}
+
+        void operator()() {
+            sleep(1);
+            _is_finish = 1;
+        }
+
+        void cancel() {}
+
+        void on_timeout() {}
+
+        char& _is_finish;
+    };
+
+    v8_engine::executor test_executor(1, ss::smp::count);
+
+    const size_t futures_count = 5;
+
+    std::vector<char> finish_flags(futures_count, 0);
+    std::vector<ss::future<>> futures;
+    futures.reserve(futures_count);
+
+    for (int i = 0; i < futures_count; ++i) {
+        futures.emplace_back(test_executor.submit(
+          task_for_test(finish_flags[i]), std::chrono::milliseconds(5000)));
+    }
+
+    test_executor.stop().get();
+    auto res = ss::when_all(futures.begin(), futures.end()).get();
+
+    for (auto i = 0; i < futures_count; ++i) {
+        if (finish_flags[i] != 1) {
+            BOOST_REQUIRE_EXCEPTION(
+              res[i].get(),
+              ss::gate_closed_exception,
+              [](const ss::gate_closed_exception& e) {
+                  return "gate closed" == std::string(e.what());
+              });
+        }
+    }
+}
```

Comment:
please configure your text editor to add newlines automatically

Reply:
done